chain/chaintest/balance_handler.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/codec/codectest"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/state/tstate"
+)
+
+// TestBalanceHandler tests b by requiring that it upholds the invariants
+// described in the BalanceHandler interface.
+func TestBalanceHandler(ctx context.Context, t *testing.T, b chain.BalanceHandler) {
+	// Test addresses
+	addrOne := codectest.NewRandomAddress()
+	addrTwo := codectest.NewRandomAddress()
+
+	tests := []struct {
+		Name string
+
+		Actor  codec.Address
+		Amount uint64
+
+		// State keys are not enforced in Setup
+		Setup func(context.Context, *testing.T, state.Mutable)
+		// State keys are enforced in TestFunc
+		TestFunc func(context.Context, codec.Address, state.Mutable, uint64) error
+		// Assertion is run after TestFunc for any additional checks
+		// State keys are not enforced in Assertion
+		Assertion func(context.Context, *testing.T, state.Mutable)
```

Comment:
Table-driven tests are great for when you're trying to just run variations of arguments against a function, e.g:
```
if tt.wantSum != Add(tt.a, tt.b) {
    t.Fatal("wrong result!")
}
```

But in this example where we're writing complex setup code/assertion code in the test vector, there isn't much of a difference between just having separate tests entirely. I think it would actually be much cleaner to just have separate calls to `t.Run` for each test individually instead of trying to force them all to fit the same test-table.

Reply:
I agree; switching to `t.Run` reduced the code by 25% 😄